charms/jupyter-controller/src/charm.py

```diff
@@ -206,6 +210,23 @@ def _apply_k8s_resources(self, force_conflicts: bool = False) -> None:
                 raise GenericCharmRuntimeError("CRD resources creation failed") from error
         self.model.unit.status = MaintenanceStatus("K8S resources created")
 
+    def _check_container_connection(self):
+        """Check if connection can be made with container."""
+        if not self.container.can_connect():
+            raise ErrorWithStatus("Pod startup is not complete", MaintenanceStatus)
+
+    def _check_status(self):
+        """Check status of workload and set status accordingly."""
+        container = self.unit.get_container(self._container_name)
+        check = container.get_check("jupyter-controller-up")
+        if check.status != CheckStatus.UP:
+            self.logger.error(
+                f"Container {self._container_name} failed health check. It will be restarted."
```

Comment:
Container will be restarted. So, message will be:
`Container jupyter-controller failed health check. It will be restarted.`
Do you think this is clear?